enumerables.rb

```diff
@@ -0,0 +1,166 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |x|
+      yield(self[x])
+    end
+    self
+  end
+
+  def my_each_with_index(value = 0)
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = value
+    size.times do |x|
+      yield(self[x], index)
+      index += 1
+    end
+  end
```

Comment:
# my_each_with_index
1. [ ] When a block is given when self is an array returns the array itself after calling the given block once for each element in self
### Test 👉 `array.my_each_with_index(&block)`

|Expected|Got|
|:---:|:---:|
|`[6, 0, 4, 8, 7, 7, 5, 7, 1, 1, 4, 7, 8, 7, 6, 3, 2, 8, 0, 2, 7, 4, 0, 4, 0, 7, 3, 6, 7, 2, 3, 4, 4, 3..., 6, 3, 5, 6, 6, 5, 1, 0, 7, 8, 5, 1, 6, 2, 4, 5, 5, 8, 8, 3, 2, 2, 3, 6, 3, 8, 8, 7, 1, 1, 4, 8, 5]`|`100`|
# 
2. [ ] When a block is given when self is a range returns the range itself after calling the given block once for each element in self
### Test 👉 `range.my_each_with_index(&block)`
Resulted in *NoMethodError*
#
3. When a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self
### Test 👉 `enum.my_each_with_index(&block)`
Resulted in *NoMethodError*
#
4. When a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self
### Test 👉 `my_each_with_index_output`

|Expected|Got|
|:---:|:---:|
|`"Num: [:a, 1], idx: 0\nNum: [:b, 2], idx: 1\nNum: [:c, 3], idx: 2\nNum: [:d, 4], idx: 3\nNum: [:e, 5], idx: 4\n"`|`"Num: , idx: 0\nNum: , idx: 1\nNum: , idx: 2\nNum: , idx: 3\nNum: , idx: 4\n"`|
#